### PR TITLE
Add profiler tests to the runtime repo

### PR DESCRIPTION
**Pull request by davmason**
Ports GCBasic, ReJIT, and EventPipe tests as well as adds a test that verifies profilers can call MetaDataGetDispenser.

### CODE REVIEW

File: src/coreclr/tests/src/profiler/common/ProfilerControl.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+using System;
```
Comment (jkotas) [inline src/coreclr/tests/src/profiler/common/ProfilerControl.cs:1]: Missing license header

File: src/coreclr/tests/src/profiler/native/classfactory.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
```
Comment (jkotas) [inline src/coreclr/tests/src/profiler/native/classfactory.cpp:1]: We should use the repo standard license header: https://github.com/dotnet/runtime/blob/master/CONTRIBUTING.md#file-headers

File: test_repro.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (noahfalk) [inline test_repro.sh:1]: was this intended to be checked in? it is at the root of the repo so we probably don't want it there.

**Review by noahfalk — APPROVED** [review]
This looks good to me David : ) I'd recommend adding a little README that describes the basics of how these tests work and how someone could run one in isolation if they need to.

File: test_repro.sh (hunk 3 above)
Comment (davmason) [reply test_repro.sh:1]: whoops, no it was not intended to be checked in

**Pull request merged**